### PR TITLE
Fixes powerloader clamp issues

### LINKED ISSUE #2383: Crates empty when picked up
## Testmerges:

#2378 #2368 #2128 #2306 #2301 #2196 #2264 #2246 #2224 #2333 #2310 #2354 #2361 #2366 #2370

## Reproduction:

Filled a crate with supply manifests. When I picked it up with a powerloader, it picked up the crate but left the supply manifests

### PR DESCRIPTION
**Pull request by Rohesie**
Fixes #2383 
Cleans up code a tiny bit.

### CODE REVIEW

File: code/modules/vehicles/powerloader.dm [hunk 1]
```diff
@@ -161,60 +161,58 @@
...
-				loaded.forceMove(T)
-				loaded = null
-				playsound(src, 'sound/machines/hydraulics_1.ogg', 40, 1)
-				update_icon()
+		if(!isturf(target))
+			return
+		var/turf/T = target
+		if(T.density)
+			return
+		for(var/atom/movable/AM in T.contents)
```
Comment (DominikPanic) [inline code/modules/vehicles/powerloader.dm]: Contents should already be atom/movable

File: code/modules/vehicles/powerloader.dm [hunk 2]
```diff
@@ -161,60 +161,58 @@
...
+			if(AM.density)
+				to_chat(user, "<span class='warning'>You can't drop [loaded] here, [AM] blocks the way.</span>")
+				return
+		if(loaded.bound_height > 32)
+			var/turf/next_turf = get_step(T, NORTH)
+			if(next_turf.density)
+				to_chat(user, "<span class='warning'>You can't drop [loaded] here, something blocks the way.</span>")
+				return
+			for(var/atom/movable/AM in next_turf.contents)
```
Comment (DominikPanic) [inline code/modules/vehicles/powerloader.dm]: same

File: code/modules/vehicles/powerloader.dm [hunk 3]
```diff
@@ -161,60 +161,58 @@
...
+				if(AM.density)
+					to_chat(user, "<span class='warning'>You can't drop [loaded] here, [AM] blocks the way.</span>")
+					return
+		if(loaded.bound_width > 32)
+			var/turf/next_turf = get_step(T, EAST)
+			if(next_turf.density)
+				to_chat(user, "<span class='warning'>You can't drop [loaded] here, something blocks the way.</span>")
+				return
+			for(var/atom/movable/AM in next_turf.contents)
```
Comment (DominikPanic) [inline code/modules/vehicles/powerloader.dm]: same

File: code/modules/vehicles/powerloader.dm (hunk 1 above)
Comment (spookydonut) [reply code/modules/vehicles/powerloader.dm]: Yeah this can be made typeless

**Pull request merged**